### ISSUE TITLE
Bug: debug mode doesn't work with structlog 

### ISSUE THREAD

**Issue by doshitan** [litestar-org/litestar#1672]
### Description

Using the default `StructLoggingConfig` for logging causes the application to fail to start with `litestar run --debug`.

### MCVE

```python
from litestar import Litestar, get
from litestar.logging import StructLoggingConfig


@get("/")
def hello_world() -> dict[str, str]:
    """Keeping the tradition alive with hello world."""
    return {"hello": "world"}

app = Litestar(
    route_handlers=[hello_world],
    logging_config=StructLoggingConfig(),
)
```

### Steps to reproduce

```bash
1. Attempt to run MCVE with `litestar run --debug`
```

### Logs

```bash
litestar run --debug
Traceback (most recent call last):
  File "/nix/store/fwh99risihjp36ypf0hjsf2lj37qiq4s-python3.11-litestar-2.0.0a6/bin/.litestar-wrapped", line 6, in <module>
    from litestar import __main__
  File "/nix/store/fwh99risihjp36ypf0hjsf2lj37qiq4s-python3.11-litestar-2.0.0a6/lib/python3.11/site-packages/litestar/__main__.py", line 3, in <module>
    litestar_group()  # pragma: no cover
    ^^^^^^^^^^^^^^^^
  File "/nix/store/ahk8gvfqrj94qg6ndvg2akn74rhp1xdz-python3.11-click-8.1.3/lib/python3.11/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/ahk8gvfqrj94qg6ndvg2akn74rhp1xdz-python3.11-click-8.1.3/lib/python3.11/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/nix/store/ahk8gvfqrj94qg6ndvg2akn74rhp1xdz-python3.11-click-8.1.3/lib/python3.11/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/ahk8gvfqrj94qg6ndvg2akn74rhp1xdz-python3.11-click-8.1.3/lib/python3.11/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/ahk8gvfqrj94qg6ndvg2akn74rhp1xdz-python3.11-click-8.1.3/lib/python3.11/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/ahk8gvfqrj94qg6ndvg2akn74rhp1xdz-python3.11-click-8.1.3/lib/python3.11/site-packages/click/decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/fwh99risihjp36ypf0hjsf2lj37qiq4s-python3.11-litestar-2.0.0a6/lib/python3.11/site-packages/litestar/cli/_utils.py", line 202, in wrapped
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/fwh99risihjp36ypf0hjsf2lj37qiq4s-python3.11-litestar-2.0.0a6/lib/python3.11/site-packages/litestar/cli/commands/core.py", line 81, in run_command
    app.debug = True
    ^^^^^^^^^
  File "/nix/store/fwh99risihjp36ypf0hjsf2lj37qiq4s-python3.11-litestar-2.0.0a6/lib/python3.11/site-packages/litestar/app.py", line 455, in debug
    self.logger.setLevel(logging.DEBUG if value else logging.INFO)
    ^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/8ynk4p5bjm1yk67vkamjy5wslgls2m2j-python3.11-structlog-23.1.0/lib/python3.11/site-packages/structlog/_config.py", line 401, in __getattr__
    return getattr(bl, name)
           ^^^^^^^^^^^^^^^^^
AttributeError: 'BoundLoggerFilteringAtInfo' object has no attribute 'setLevel'
```

### Litestar Version

2.0.0a6

### Platform

- [X] Linux
- [ ] Mac
- [ ] Windows
- [ ] Other (Please specify in the description above)

---
> [!NOTE]  
> While we are open for sponsoring on [GitHub Sponsors](https://github.com/sponsors/litestar-org/) and 
> [OpenCollective](https://opencollective.com/litestar), we also utilize [Polar.sh](https://polar.sh/) to engage in pledge-based sponsorship.
>
> Check out all issues funded or available for funding [on our Polar.sh dashboard](https://polar.sh/litestar-org)
> * If you would like to see an issue prioritized, make a pledge towards it!
> * We receive the pledge once the issue is completed & verified
> * This, along with engagement in the community, helps us know which features are a priority to our users.

<a href="https://polar.sh/litestar-org/litestar/issues/1672">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/litestar-org/litestar/issues/1672/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/litestar-org/litestar/issues/1672/pledge.svg">
</picture>
</a>

**Comment by JacobCoffee** [comment]
Structlogs concept of log levels is weird.
We have a default wrapper class (litestar.logging.config.default_wrapper_class) 

We could check it/override there 
```py
wrapper_class=structlog.make_filtering_bound_logger(...log level here...),
```

I was able to modify our debug setter in `litestar.app.Litestar.debug`

```py
@debug.setter
    def debug(self, value: bool) -> None:
        try:
            import structlog

            if isinstance(self.logger, structlog.BoundLogger):
                if value: 
                    new_filter = structlog.stdlib.filter_by_level(DEBUG)
                else: 
                    new_filter = structlog.stdlib.filter_by_level(INFO)
                self.logger = new_filter(self.logger._logger)
        except ImportError:
            # Fallback to standard library logger
            self.logger.setLevel(logging.DEBUG if value else logging.INFO)

            if isinstance(self.logging_config, LoggingConfig):
                self.logging_config.loggers["litestar"]["level"] = "DEBUG" if value else "INFO"
            self._debug = value
```

This allows it to run. rfc @litestar-org/maintainers

**Comment by Goldziher** [comment]
> Structlogs concept of log levels is weird.
> We have a default wrapper class (litestar.logging.config.default_wrapper_class) 
> 
> We could check it/override there 
> ```py
> wrapper_class=structlog.make_filtering_bound_logger(...log level here...),
> ```
> 
> I was able to modify our debug setter in `litestar.app.Litestar.debug`
> 
> ```py
>     @debug.setter
>     def debug(self, value: bool) -> None:
>         try:
>             import structlog
> 
>             if isinstance(self.logger, structlog.BoundLogger):
>                 if value: 
>                     new_filter = structlog.stdlib.filter_by_level(DEBUG)
>                 else: 
>                     new_filter = structlog.stdlib.filter_by_level(INFO)
>                 self.logger = new_filter(self.logger._logger)
>         except ImportError:
>             # Fallback to standard library logger
>             self.logger.setLevel(logging.DEBUG if value else logging.INFO)
> 
>             if isinstance(self.logging_config, LoggingConfig):
>                 self.logging_config.loggers["litestar"]["level"] = "DEBUG" if value else "INFO"
>             self._debug = value
> ```
> ```py
> from litestar import Litestar, Request, get
> from litestar.logging import StructLoggingConfig
> 
> @get("/")
> def hello_world(request: Request) -> dict[str, str]:
>     """Keeping the tradition alive with hello world."""
>     request.logger.info("inside a request")
>     return {"hello": "world"}
> 
> 
> app = Litestar(
>     route_handlers=[hello_world],
>     logging_config=StructLoggingConfig(),
> )
> ```
> ```console
> git/testing/litestar is  v0.1.0 via  v3.11.2 (testing-py3.11) 🐠 took 1m25s
> ➜ litestar run --debug
> Using Litestar app from app:app
> ┌──────────────────┬──────────────────────┐
> │ Litestar version │ 2.0.0                │
> │ Debug mode       │ Enabled              │
> │ CORS             │ Disabled             │
> │ CSRF             │ Disabled             │
> │ OpenAPI          │ Enabled path=/schema │
> │ Compression      │ Disabled             │
> └──────────────────┴──────────────────────┘
> Starting server process ─────────────────────────────────────────────────────────────
> INFO:     Started server process [151775]
> INFO:     Waiting for application startup.
> INFO:     Application startup complete.
> INFO:     Uvicorn running on http://127.0.0.1:8000 (Press CTRL+C to quit)
> {"event":"inside a request","level":"info","timestamp":"2023-05-13T21:47:58.078775Z"}
> INFO:     127.0.0.1:46340 - "GET / HTTP/1.1" 200 O
> ```
> 
> This allows it to run but im unsure of its correctness so -> rfc @litestar-org/maintainers 

Make a PR?

**Comment by JacobCoffee** [comment]
Sure i can do that! I just wasn't sure if my approach was correct 😛

**Comment by Goldziher** [comment]
> Sure i can do that! I just wasn't sure if my approach was correct 😛 

I think the only one with significant experience is @peterschutt

**Comment by provinzkraut** [comment]
@JacobCoffee that solution looks good to me, but I'd prefer not having to go through all of the `try/import/except` thing and handling for structlog within the setter. 

How about we simply create a `set_log_level` wrapper, which implements your solution for structlog and the standard way for everything else, and then we simply detect structlog once after the logger has been configured, and set the appropriate callable, which the setter can then refer to?

That would also be a good future-proofing, since it'd be easier to add other logging libs which might deviate from the stdlib's API.

**Comment by JacobCoffee** [comment]
Sounds good to me

**Comment by JacobCoffee** [comment]
I'll be trying to work on this one today after I wake up

**Comment by JacobCoffee** [comment]
With the latest changes from main it seems this no longer errors out. Probably from #1742..

but, I am thinking we should set any logger to debug level if `--debug` is passed via cli

**Comment by JacobCoffee** [comment]
I am closing this as resolved since the error is now gone in main thanks to #1742 and opening up a more precise issue around the fact that logging does not properly cascade the filter levels when `--debug` is set.

https://github.com/litestar-org/litestar/issues/1804

**Comment by cofin** [comment]
Resolved by #2943

**Comment by cofin** [comment]
Resolved by #2943
